File: Android/indoor_localization/lib/entities/asset_position_history_log_filter.dart
class AssetPositionHistoryLogFilter {
  final int facilityId;
  final int assetId;
  final DateTime startDate;
  final DateTime endDate;

  AssetPositionHistoryLogFilter({
    required this.facilityId,
    required this.assetId,
    required this.startDate,
    required this.endDate,
  });
}

File: WebApp/indoor-localization/src/entities/AssetPositionHistoryLogFilter.ts
export class AssetPositionHistoryLogFilter {
  facilityId: number | null;
  assetId: number | null;
  startDate: Date | null;
  endDate: Date | null;

  constructor(
    facilityId: number | null,
    assetId: number | null,
    startDate: Date | null,
    endDate: Date | null
  ) {
    this.facilityId = facilityId;
    this.assetId = assetId;
    this.startDate = startDate;
    this.endDate = endDate;
  }
}
